packages/expo/cli/start/server/middleware/ClassicManifestMiddleware.ts

```diff
@@ -0,0 +1,159 @@
+import { ExpoAppManifest, ExpoConfig } from '@expo/config';
+import chalk from 'chalk';
+import os from 'os';
+
+import { APISettings } from '../../../api/settings';
+import { signClassicExpoGoManifestAsync } from '../../../api/signManifest';
+import UserSettings from '../../../api/user/UserSettings';
+import { ANONYMOUS_USERNAME, getUserAsync } from '../../../api/user/user';
+import * as Log from '../../../log';
+import { logEvent } from '../../../utils/analytics/rudderstackClient';
+import { memoize } from '../../../utils/fn';
+import { learnMore } from '../../../utils/link';
+import { stripPort } from '../../../utils/url';
+import { DEVELOPER_TOOL, HostInfo, ManifestMiddleware, ParsedHeaders } from './ManifestMiddleware';
+import { assertRuntimePlatform, parsePlatformHeader } from './resolvePlatform';
+import { ServerHeaders, ServerRequest } from './server.types';
+
+type SignManifestProps = {
+  manifest: ExpoAppManifest;
+  hostId: string;
+  acceptSignature: boolean;
+};
+
+export class ClassicManifestMiddleware extends ManifestMiddleware {
+  public getParsedHeaders(req: ServerRequest): ParsedHeaders {
+    const platform = parsePlatformHeader(req) || 'ios';
+    assertRuntimePlatform(platform);
+    return {
+      platform,
+      acceptSignature: Boolean(req.headers?.['exponent-accept-signature']),
+      hostname: stripPort(req.headers?.['host']),
```

Comment:
req.headers is always non-null here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e7f55d3765c3ad57a19f7e47cc8e8ca9faca9fc4/types/node/http.d.ts#L871